<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Tipo-Test" data-toc-modified-id="Tipo-Test-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Tipo Test</a></span></li></ul></div>

## Tipo Test

In [14]:
import pandas as pd
import numpy as np
import edhec_risk_kit as erk
import matplotlib.pyplot as plt
import plotly
import plotly.graph_objs as go
from plotly.offline import init_notebook_mode, iplot
from plotly.subplots import make_subplots
import warnings

# Configuración de librerías
init_notebook_mode(connected=True)  
warnings.filterwarnings("ignore")
# Modo desarrollo del módulo edhec_risk_kit
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


__Lea los datos en el archivo "Portfolios_Formed_on_ME_monthly_EW.csv" Realizamos una serie de análisis en las columnas 'Lo 10' y 'Hi 10' que son los rendimientos de las carteras del decil más bajo y más alto. Para esta tarea, utilizaremos las carteras del quintil más bajo y más alto, que se denominan "Lo 20" y "Hi 20" respectivamente.__

In [15]:
df_ffme = erk.get_ffme_returns(directory='data')
df_ffme.head()

SmallCap  LargeCap
1926-07   -0.0057    0.0333
1926-08    0.0384    0.0233
1926-09   -0.0048   -0.0009
1926-10   -0.0329   -0.0295
1926-11   -0.0055    0.0316

__Pregunta 1. ¿Cuál fue el rendimiento anualizado de la cartera Lo 20 durante todo el período?__

Respuesta: 15.20%

In [17]:
n_months = df_ffme.shape[0]
return_per_month = (df_ffme+1).prod()**(1/n_months) - 1
annualized_return = (return_per_month + 1)**12-1
print('Annualized returns:',annualized_return,sep=2*'\n')

Annualized returns:

SmallCap    0.151977
LargeCap    0.098490
dtype: float64


__Pregunta 2 ¿Cuál fue la volatilidad anualizada de la cartera Lo 20 durante todo el periodo?__

Respuesta: 33.67%

In [19]:
# Calcular volatilidad anualizada para comparar ambos activos
annualized_vol = df_ffme.std()*np.sqrt(12)
print('Volatilidad anualizada',annualized_vol,sep='\n')

Volatilidad anualizada
SmallCap    0.336701
LargeCap    0.195116
dtype: float64


__Pregunta 3 ¿Cuál fue la rentabilidad anualizada de la cartera Hi 20 durante todo el periodo?__

Respuesta 9.85%

__Pregunta 4 ¿Cuál fue la volatilidad anualizada de la cartera Hi 20 durante todo el periodo?__

Respuesta: 19.51%

__Pregunta 5 ¿Cuál fue la rentabilidad anualizada de la cartera de Lo 20 durante el periodo 1999 - 2015 (ambos inclusive)?__

Respuesta: 11.44%


In [34]:
df = df_ffme["1999":"2015"]

n_months = df.shape[0]
return_per_month = (df+1).prod()**(1/n_months) - 1
annualized_return = (return_per_month + 1)**12-1
print('Annualized returns:',annualized_return,sep=2*'\n')

Annualized returns:

SmallCap    0.114426
LargeCap    0.062865
dtype: float64


__Pregunta 6 ¿Cuál fue la volatilidad anualizada de la cartera de Lo 20 durante el periodo 1999 - 2015 (ambos inclusive)?__

Respuesta: 22.89%

In [35]:
# Calcular volatilidad anualizada para comparar ambos activos
annualized_vol = df.std()*np.sqrt(12)
print('Volatilidad anualizada',annualized_vol,sep='\n')

Volatilidad anualizada
SmallCap    0.228894
LargeCap    0.172713
dtype: float64


__Pregunta 7 ¿Cuál fue la rentabilidad anualizada de la cartera Hi 20 durante el periodo 1999 - 2015 (ambos inclusive)?__

Respuesta: 6.28%


__Pregunta 8 ¿Cuál fue la volatilidad anualizada de la cartera Hi 20 durante el periodo 1999 - 2015 (ambos inclusive)?__

Respuesta: 17.27%

__Pregunta 9 ¿Cuál fue el MaxDrowdown (expresado como un número positivo) experimentado durante el periodo 1999-2015 en la cartera SmallCap (Lo 20)?__

Respuesta: 62.48%

In [24]:
df_drawdowns = erk.drawdown(df['SmallCap'], capital_inicial=1000)
erk.plot_drawdowns(df_drawdowns, title='SmallCap')

__Pregunta 10. Al final de qué mes del período 1999-2015 se produjo esa reducción máxima en la cartera SmallCap (Lo 20)?__

Respuesta: 2009-02

__Pregunta 11. ¿Cuál fue el MaxDrowdown (expreasado como un número positivo) experimentada durante el periodo 1999-2015 en la cartera LargeCap (Hi 20)?__

Respuesta: 55.27%

In [26]:
df_drawdowns = erk.drawdown(df['LargeCap'], capital_inicial=1000)
erk.plot_drawdowns(df_drawdowns, title='LargeCap')

__Pregunta 12. Al final de qué mes del período 1999-2015 se produjo esa reducción máxima en la cartera LargeCap (Hi 20)?__

Respuesta: 2009-02

__Para las preguntas restantes, utilice el conjunto de datos de los Índices de Fondos de Cobertura EDHEC.__

In [28]:
df_hfi = erk.get_hfi_returns(directory='data/')

time1 = '2009'

df = df_hfi.loc[(df_hfi.index>=time1)].copy()

__Respuesta 13. Observando los datos desde 2009 (incluyendo todo el año 2009) hasta 2018, ¿qué índice de fondos de cobertura ha mostrado la mayor semidesviación?__

Respuesta: Short Selling

In [29]:
erk.negative_semideviation(df).sort_values(ascending=False)

Short Selling             0.021240
Emerging Markets          0.015276
Long/Short Equity         0.012139
CTA Global                0.010996
Event Driven              0.010135
Distressed Securities     0.009498
Funds Of Funds            0.007948
Convertible Arbitrage     0.006681
Relative Value            0.005711
Equity Market Neutral     0.005186
Global Macro              0.005075
Merger Arbitrage          0.003611
Fixed Income Arbitrage    0.003321
dtype: float64

__Pregunta 14. Si observamos los datos desde 2009 (incluyendo todo el año 2009), ¿qué índice de fondos de cobertura ha mostrado la menor semidesviación?__

Respuesta: Fixed Income Arbitrage

__Pregunta 15. Si se observan los datos desde 2009 (incluyendo todo el año 2009), ¿cuál es el índice de fondos de cobertura que ha tenido un valor de asimetría más negativo?__

Respuesta: Equity Market Neutral

In [30]:
erk.skewness(df).sort_values(ascending=False)

Convertible Arbitrage     1.305911
Fixed Income Arbitrage    1.121453
Short Selling             0.456518
Global Macro              0.348184
Relative Value            0.159953
CTA Global                0.052062
Emerging Markets          0.033123
Distressed Securities    -0.254944
Long/Short Equity        -0.463703
Event Driven             -0.488821
Merger Arbitrage         -0.551065
Funds Of Funds           -0.646908
Equity Market Neutral    -0.896327
dtype: float64

__Pregunta 16. Si se observan los datos desde el año 2000 (incluyendo todo el año 2000) hasta 2018, ¿qué índice de fondos de cobertura ha mostrado la mayor curtosis?__

Respuesta: Fixed Income Arbitrage

In [31]:
time1 = '2000'

df = df_hfi.loc[(df_hfi.index>=time1)].copy()
erk.kurtosis(df).sort_values(ascending=False)

Fixed Income Arbitrage    33.378729
Convertible Arbitrage     23.001449
Equity Market Neutral     20.329621
Relative Value            12.731096
Funds Of Funds             7.584725
Distressed Securities      6.231720
Emerging Markets           5.941289
Event Driven               5.543444
Merger Arbitrage           4.520785
Long/Short Equity          4.472589
Short Selling              4.352758
Global Macro               3.653278
CTA Global                 2.845022
dtype: float64